test/e2e/windows/memory_limits.go

```diff
@@ -104,32 +104,38 @@ func overrideAllocatableMemoryTest(f *framework.Framework, allocatablePods int)
 	const (
 		podType = "memory_limit_test_pod"
 	)
-
-	totalAllocatable := getTotalAllocatableMemory(f)
-
-	memValue := totalAllocatable.Value()
-	memPerPod := memValue / int64(allocatablePods)
-	ginkgo.By(fmt.Sprintf("Deploying %d pods with mem limit %v, then one additional pod", allocatablePods, memPerPod))
-
-	// these should all work
-	pods := newMemLimitTestPods(allocatablePods, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().CreateBatch(pods)
-
-	failurePods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().Create(failurePods[0])
-
-	gomega.Eventually(func() bool {
-		eventList, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{})
-		framework.ExpectNoError(err)
-		for _, e := range eventList.Items {
-			// Look for an event that shows FailedScheduling
-			if e.Type == "Warning" && e.Reason == "FailedScheduling" && e.InvolvedObject.Name == failurePods[0].ObjectMeta.Name {
-				framework.Logf("Found %+v event with message %+v", e.Reason, e.Message)
-				return true
+	//
+	selector := labels.Set{"kubernetes.io/os": "windows"}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+	framework.ExpectNoError(err)
+	totalAllocatable := resource.NewQuantity(0, resource.BinarySI)
+	for _, node := range nodeList.Items {
+		status := node.Status
+		totalAllocatable.Add(status.Allocatable[v1.ResourceMemory])
+		nodeAllocatable := totalAllocatable
+
+		pods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(nodeAllocatable, 1))
```

Comment:
We only need to create a single pod here.  The could be simplified/removed?

Reply:
@jsturtevant the value of parameter `numPods` in func `newMemLimitTestPods` is 1, so I am not understanding why IYO it is creating multiple pods. Could you explain why and how can we make it to create a single pod.

Reply:
I think the way you are usisng it is creating a single pod, the thought was maybe this isn't needed as it was designed for creating multiple pods which we aren't using any more.